app/controllers/IndexController.scala

```
@@ -17,33 +17,58 @@
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
 import uk.gov.hmrc.auth.core.AffinityGroup
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 
+import java.time.{Clock, Instant}
 import javax.inject.Inject
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 class IndexController @Inject() (
     val controllerComponents: MessagesControllerComponents,
     identify: IdentifierAction,
     checkEnrolment: CheckEnrolledToServiceAction,
-    retrieveCtUTR: CtUtrRetrievalAction
-) extends FrontendBaseController
-    with I18nSupport {
+    retrieveCtUTR: CtUtrRetrievalAction,
+    sessionRepository: SessionRepository,
+    clock: Clock
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with Logging {
 
   def onPageLoad(): Action[AnyContent] = (identify() andThen checkEnrolment andThen retrieveCtUTR()).async {
     implicit request =>
       request.affinityGroup match {
         case AffinityGroup.Individual =>
-          Future.successful(Redirect(controllers.routes.IndividualRegistrationTypeController.onPageLoad(NormalMode)))
+          Future.successful(
+            Redirect(controllers.routes.IndividualRegistrationTypeController.onPageLoad(NormalMode))
+          )
         case _                        =>
           request.utr match {
             case Some(utr) =>
-              Future.successful(Ok("User has UTR. Redirect them to Is This Your Business? page (CARF-126)"))
-            case None      =>
+              // org with CT UTR, - CT Automatched - go straight to is this your business?
+              val userAnswers = UserAnswers(request.userId, lastUpdated = Instant.now(clock))
+              for {
+                autoMatchedUserAnswers <- Future.fromTry(userAnswers.set(IndexPage, utr))
+                result                 <- sessionRepository.set(autoMatchedUserAnswers).map {
```

I don't think we need to map this here in the for comprehension. In the other controllers we let the future fail (and the user is automatically sent to an error page) if there is error setting user answers